eng/pipelines/templates/Build_and_UnitTest.yml

```diff
@@ -88,13 +96,23 @@ steps:
     configuration: "$(BuildConfiguration)"
     msbuildArguments: "/t:RestoreVS /p:BuildNumber=$(BuildNumber) /p:BuildRTM=$(BuildRTM) /v:m"
 
+- task: MSBuild@1
+  displayName: "Build for VS2019"
+  inputs:
+    solution: "build\\build.proj"
+    msbuildVersion: "16.0"
+    configuration: "$(BuildConfiguration)"
+    msbuildArguments: "/t:BuildNoVSIX /p:BuildRTM=$(BuildRTM) /p:BuildNumber=$(BuildNumber) /p:SkipILMergeOfNuGetExe=true /p:GitRepositoryRemoteName=github"
+  condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))" #run in official non-RTM builds, for generating sourcelink
+
 - task: MSBuild@1
   displayName: "Build for VS2019"
   inputs:
     solution: "build\\build.proj"
     msbuildVersion: "16.0"
     configuration: "$(BuildConfiguration)"
     msbuildArguments: "/t:BuildNoVSIX /p:BuildRTM=$(BuildRTM) /p:BuildNumber=$(BuildNumber) /p:SkipILMergeOfNuGetExe=true"
+  condition: "and(succeeded(), or(eq(variables['BuildRTM'], 'true'), eq(variables['IsOfficialBuild'], 'false')))" #run in official RTM and non-official builds
```

Comment:
I feel like there must be a better way we can add an argument in some conditions and not others, so we can avoid duplicating the entire step.

Can we set a build variable to contain the `/p:GitRepositoryRemoteName=github` in the "Add remote repository" task,  then use the in the `msbuildArguments` here? I assume the variable will expand to an empty string when not set.

Reply:
Thanks for reviewing. Just tested with your approach locally. I think you're right.
Fixed in the latest commit.